src/main/java/org/terasology/changingBlocks/conditional/BlockCondition.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.changingBlocks.conditional;
+
+import org.terasology.math.Side;
+
+public class BlockCondition {
+    /**
+     * The chance that this condition will be executed if it is reached, where 0.0 = 0% and 1.0 = 100%
+     */
+    public float chance = 1.0F;
+
+    /**
+     * The block ID of the block which will replace this block.
+     */
+    public String blockToBecome;
```

Comment:
I'm thinking of some better name to go well with `triggerBlock` and to reflect the fact that this is supposed to be an id... maybe:
```suggestion
    public String targetBlockId;
```

Reply:
Sounds good to me.